File: 03-poo/04-encapsulamiento-y-modificadores-acceso/scala/soluciones/RetoCuentaAhorro.scala
class CuentaAhorro(private var saldo: Double):
  require(saldo >= 0)
  def retirar(monto: Double): Boolean =
    if monto <= 0 || monto > saldo then false
    else { saldo -= monto; true }
  def saldoActual: Double = saldo

@main def retoCuentaAhorro(): Unit =
  val c = new CuentaAhorro(100)
  c.retirar(40)
  println(c.saldoActual)

File: 03-poo/04-encapsulamiento-y-modificadores-acceso/java/soluciones/RetoCuentaAhorro.java
public class RetoCuentaAhorro {
    static class CuentaAhorro {
        private double saldo;
        CuentaAhorro(double saldoInicial) { if (saldoInicial < 0) throw new IllegalArgumentException(); saldo = saldoInicial; }
        boolean retirar(double monto) {
            if (monto <= 0 || monto > saldo) return false;
            saldo -= monto;
            return true;
        }
        double getSaldo() { return saldo; }
    }
    public static void main(String[] args) {
        CuentaAhorro cuenta = new CuentaAhorro(100);
        cuenta.retirar(40);
        System.out.println(cuenta.getSaldo());
    }
}
